xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
The difference here (and that we need to take care). It takes its own lock
... so all the other stuff which now was "locked out" by the general
m_critSection will continue to run now :-) So careful, please.

2016-04-01 18:01 GMT+02:00 Pär Björklund <notifications@github.com>:

> In xbmc/peripherals/bus/PeripheralBus.h
> <https://github.com/xbmc/xbmc/pull/9510#discussion_r58227335>:
>
> > @@ -54,7 +54,7 @@ namespace PERIPHERALS
> >      /*!
> >       * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
> >       */
> > -    bool NeedsPolling(void) const { return m_bNeedsPolling; }
> > +    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
>
> It's the same basic complexity under the hood, a lock will likely be
> implemented using cas, switching to an atomic makes the visible parts
> simpler as the code would be return m_bNeedsPolling with no visible
> synchronization logic cluttering up the code
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/9510/files/dbc772050041cf080e49f84b42dbec57ec3862df..a3b9caf0a888c4dcebf008416f6aab77ffbd8117#r58227335>
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B